admin/server/organizations.go

```diff
@@ -34,13 +54,23 @@ func (s *Server) GetOrganization(ctx context.Context, req *adminv1.GetOrganizati
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	if !claims.CanOrganization(ctx, org.ID, auth.ReadOrg) {
+		return nil, status.Error(codes.PermissionDenied, "not allowed to read org")
+	}
+
 	return &adminv1.GetOrganizationResponse{
 		Organization: orgToDTO(org),
 	}, nil
 }
 
 func (s *Server) CreateOrganization(ctx context.Context, req *adminv1.CreateOrganizationRequest) (*adminv1.CreateOrganizationResponse, error) {
-	org, err := s.admin.DB.InsertOrganization(ctx, req.Name, req.Description)
+	// Check the request is made by an authenticated user
+	claims := auth.GetClaims(ctx)
+	if claims.OwnerType() != auth.OwnerTypeUser {
+		return nil, status.Error(codes.Unauthenticated, "not authenticated as a user")
+	}
```

Comment:
No need to check it's a user unless using `auth.OwnerID()`. Can just rely on the `Can...` check